client/src/components/PacmanGame/PacmanGame.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  initSquareGridState, getFoods, getGhosts,
+  getPacman, getWalls, getEnergizers,
+  entityToCode, entityApplier,
+} from './gameCore';
+import PacmanBoard from './PacmanBoard';
+
+class PacmanGame extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      pacman: {
+        x: 0,
+        y: 5,
+      },
+      direction: 'RIGHT',
+      score: 0,
+      gridState: [],
+      config: {
+        refreshRate: 50
+      }
+    };
+  }
+
+  setInitialGameState = () => {
+    const { width, numberofCells } = this.props;
+    const canvasWidth = width;
+    const cellsInEachRow = numberofCells;
+    // const canvasWidth = this.props.width;
+    // const cellsInEachRow = this.props.numberofCells;
+    const gridSize = canvasWidth / cellsInEachRow;
+    const gridState = initSquareGridState(cellsInEachRow);
+
+    const entitiesLocation = {
+      food: getFoods(),
+      pacman: getPacman(),
+      wall: getWalls(cellsInEachRow),
+      ghost: getGhosts(),
+      energizer: getEnergizers(),
+    };
+
+    const entitiesName = Object.keys(entitiesLocation);
+
+    entitiesName.map(entityName => entityApplier(gridState,
+      entitiesLocation[entityName],
+      entityToCode(entityName)));
+
+    this.setState({ gridState, gridSize });
+
+    let walls = [];
+    gridState.forEach((rowArray, rowIndex) => {
+      rowArray.forEach((cell, columnIndex) => {
+        if (cell === 4) {
+          walls.push([rowIndex, columnIndex]);
+        }
+      })
+    });
+    this.setState({ walls });
+  }
+  startGame = () => {
+    this.animationHandler = setInterval(
+      this.animateGame,
+      this.state.config.refreshRate
+    );
+    document.addEventListener('keydown', this.setDirection);
+  }
+  checkCollision = ({ x, y }) => {
+    let walls = this.state.walls;
+    let hasCollision = false;
+
+    walls.forEach((wall) => {
```

Comment:
Use .find for checking this

Reply:
This logic will be replaced in the next PR.

Reply:
why?